### PR TITLE
New TensorboardLogger, batch2device fix, FilteringCollateFn, minor refactorings.

### PR DESCRIPTION
**Pull request by hexfaker**

### CODE REVIEW

File: data/collate_fn.py [hunk 1]
```diff
@@ -48,3 +51,26 @@ def stack_non_text_features(data, key):
     }
 
     return data
+
+class FilteringCollateFn():
```
Comment (Scitator) [inline data/collate_fn.py]: brackets?

File: dl/callbacks.py [hunk 2]
```diff
@@ -165,6 +167,56 @@ def on_loader_end(self, state):
...
+        logger = SummaryWriter(log_dir)
+        return logger
+
+
+class GroupTensorboardLogger(Callback):
+    # formatter:off
+    def __init__(self,
+            logdir,
+            metric_names: List[str] = None,
+            log_on_batch_end=False
```
Comment (Scitator) [inline dl/callbacks.py]: why don't you add 2 flags? for `batch_end` and `epoch_end`?

File: dl/state.py [hunk 3]
```diff
@@ -9,16 +9,22 @@ class RunnerState(FrozenClass):
     An object that is used to pass internal state during train/valid/infer.
     """
 
-    def __init__(self, **kwargs):
+    def __init__(self,
```
Comment (Scitator) [inline dl/state.py]: self to next below pls

**Comment by Scitator** [comment]
@hexfaker Please, check out the CI:
https://travis-ci.com/Scitator/catalyst/jobs/157277820

File: data/collate_fn.py (hunk 1 above)
Comment (hexfaker) [reply data/collate_fn.py]: Done

File: dl/callbacks.py (hunk 2 above)
Comment (hexfaker) [reply dl/callbacks.py]: Looks like my solution is more flexible. You can log some metrics on batches, some on epochs, and some on both while using two instances of this callback. Supporting same functionality in one instance would introduce unneeded complexity to code of this callback. Now it's simple and quite readable, I'd prefer to keep it that way.

File: dl/state.py (hunk 3 above)
Comment (hexfaker) [reply dl/state.py]: done

**Comment by hexfaker** [comment]
@Scitator Yes I did. Most errors produced by lines I even haven't modified. So...

File: dl/callbacks.py (hunk 2 above)
Comment (Scitator) [reply dl/callbacks.py]: Yup, `You can log some metrics on batches, some on epochs` using only **one** instance of callback. I think, it is much easier to think from metrics perspective. So, log some metric only on batch, some only on epoch, another one – both.

Comment (hexfaker) [reply dl/callbacks.py]: @Scitator Same functionality with more code is not good.

Comment (Scitator) [reply dl/callbacks.py]: Agree, thats why I don't like **two** instances approach.

**Comment by Scitator** [comment]
Thanks! Merged.

**Pull request merged**